### PR TITLE
rgw: multisite: fix new namesake-file sync fail when disable versioning

### PR DESCRIPTION
**Pull request by dongbula**
fix http://tracker.ceph.com/issues/18426

Signed-off-by: lvshuhua <lvshuhua@cmss.chinamobile.com>

### CODE REVIEW

**Comment by dongbula** [comment]
https://github.com/ceph/ceph/pull/12932

**Comment by cbodley** [comment]
[teuthology run](http://pulpito.ceph.com/cbodley-2017-02-15_16:05:58-rgw-wip-cbodley-testing---basic-mira/) was clean (only failures were selinux related)

can you please rebase to resolve conflicts?

**Comment by dongbula** [comment]
@cbodley done it

File: src/rgw/rgw_rados.cc [hunk 1]
```diff
@@ -6468,6 +6468,8 @@ int RGWRados::Object::Write::_do_write_meta(uint64_t size, uint64_t accounted_si
 
   bool versioned_op = (target->versioning_enabled() || is_olh || versioned_target);
 
+  struct rgw_bucket_dir_entry_meta dir_meta;
```
Comment (amitkumar50) [inline src/rgw/rgw_rados.cc]: @dongbula (NIT) Kindly consider to keep PR title < 50 charactershttps://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#title-of-pull-requests-and-title-of-commits

Comment (dongbula) [inline src/rgw/rgw_rados.cc]: Well, I will update that, thanks

Comment (amitkumar50) [inline src/rgw/rgw_rados.cc]: @dongbula Great thanks

**Pull request closed without merging**